### PR TITLE
allow reservoir to be live configured

### PR DESCRIPTION
**Pull request by rf**
@Raynos 

the way I return both parts of the logger is pretty gross, if you can think of a cleaner way to do that it would be appreciated

### CODE REVIEW

File: hyperbahn/clients/index.js [hunk 1]
```diff
@@ -300,6 +307,24 @@ ApplicationClients.prototype.onRemoteConfigUpdate = function onRemoteConfigUpdat
     self.updateExemptServices();
     self.updateRpsLimitForServiceName();
     self.updateKValues();
+    self.updateLogReservoirSize();
+};
+
+ApplicationClients.prototype.updateReservoir = function updateReservoir() {
+    var self = this;
+    if (self.logReservoir) {
```
Comment (Raynos) [inline hyperbahn/clients/index.js]: `self.reservoir` ?

**Comment by Raynos** [comment]
Too much complexity here.

Just always use the reservior and have it always enable; lets toggle the flushInterval & size by flipr though.

**Comment by rf** [comment]
okay

File: hyperbahn/clients/logger.js [hunk 2]
```diff
@@ -104,5 +104,8 @@ function createLogger(options) {
         statsd: options.statsd
     });
 
-    return logger;
+    return {
```
Comment (Raynos) [inline hyperbahn/clients/logger.js]: consider

```
logger.reservoir = larchBackend;
return logger
```

**Comment by Raynos** [comment]
legit.

File: hyperbahn/clients/index.js [hunk 3]
```diff
@@ -300,6 +307,18 @@ ApplicationClients.prototype.onRemoteConfigUpdate = function onRemoteConfigUpdat
     self.updateExemptServices();
     self.updateRpsLimitForServiceName();
     self.updateKValues();
+    self.updateReservoir();
+};
+
+ApplicationClients.prototype.updateReservoir = function updateReservoir() {
+    var self = this;
+    if (self.logReservoir) {
+        var size = self.remoteConfig.get('log.reservoir.size', 100);
```
Comment (Raynos) [inline hyperbahn/clients/index.js]: Wait we made a mistake here; we do not want the default to be `100`; let's bump the default to `1000`

**Pull request merged**